### PR TITLE
VReplication DryRun: Report current dry run results for v2 commands

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
Signed-off-by: Rohit Nayak <rohit@planetscale.com>

## Description
* Implemented -dry_run flag for v2 commands
* Essentially the same dry run logs from v1 commands are reported with minor changes

## Related Issue(s)
#7071  

## Checklist
- [ ] Should this PR be backported?
- [X] Tests were added or are not required
- [ ] Documentation was added or is not required

## Impacted Areas in Vitess
Components that this PR will affect:

- [ ]  Query Serving
- [X]  VReplication
- [ ]  Cluster Management
- [ ]  Build

### CODE REVIEW

File: go/test/endtoend/vreplication/vreplication_test_env.go [hunk 1]
```diff
@@ -26,16 +26,7 @@ var dryRunResultsSwitchWritesCustomerShard = []string{
...
-	"       customer.customer@rdonly => customer.customer",
-	"       customer.customer@replica => customer.customer",
-	"       customer@rdonly => customer.customer",
-	"       customer@replica => customer.customer",
-	"       product.customer@rdonly => customer.customer",
-	"       product.customer@replica => customer.customer",
-	"Following rules will be added:",
-	"       customer => customer.customer",
-	"       product.customer => customer.customer",
+	"Routing rules for tables customer will be updated",
```
Comment (deepthi) [inline go/test/endtoend/vreplication/vreplication_test_env.go:29]: Minor formatting suggestion:
```
Routing rules for tables [customer] will be updated
```

File: go/test/endtoend/vreplication/vreplication_test_env.go [hunk 2]
```diff
@@ -50,7 +41,8 @@ var dryRunResultsSwitchWritesCustomerShard = []string{
 
 var dryRunResultsReadCustomerShard = []string{
 	"Lock keyspace product",
-	"Switch reads for tables customer to keyspace customer",
+	"Switch reads for tables customer to keyspace customer for tablet types REPLICA",
```
Comment (deepthi) [inline go/test/endtoend/vreplication/vreplication_test_env.go:44]: Similarly here:
```
... [customer] ... [REPLICA]
```

**Review by deepthi — COMMENTED** [review]
Looks fine to me but we should wait for @sougou's review

File: go/test/endtoend/vreplication/vreplication_test_env.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/test/endtoend/vreplication/vreplication_test_env.go]: fixed

File: go/test/endtoend/vreplication/vreplication_test_env.go (hunk 2 above)
Comment (rohit-nayak-ps) [reply go/test/endtoend/vreplication/vreplication_test_env.go]: fixed

File: go/vt/vtctl/vtctl.go [hunk 3]
```diff
@@ -1968,7 +1968,7 @@ func commandVRWorkflow(ctx context.Context, wr *wrangler.Wrangler, subFlags *fla
 	workflowType wrangler.VReplicationWorkflowType) error {
 
 	cells := subFlags.String("cells", "", "Cell(s) or CellAlias(es) (comma-separated) to replicate from.")
-	tabletTypes := subFlags.String("tablet_types", "", "Source tablet types to replicate from (e.g. master, replica, rdonly). Defaults to -vreplication_tablet_type parameter value for the tablet, which has the default value of replica.")
+	tabletTypes := subFlags.String("tablet_types", "master,replica,rdonly", "Source tablet types to replicate from (e.g. master, replica, rdonly). Defaults to -vreplication_tablet_type parameter value for the tablet, which has the default value of replica.")
 	dryRun := subFlags.Bool("dry_run", false, "Does a dry run of SwitchReads and only reports the actions to be taken")
```
Comment (ajm188) [inline go/vt/vtctl/vtctl.go:1972]: This help should be updated to reflect line 2146

**Review by ajm188 — APPROVED** [review]
very minor helptext change, otherwise lg2m

File: go/vt/vtctl/vtctl.go (hunk 3 above)
Comment (rohit-nayak-ps) [reply go/vt/vtctl/vtctl.go]: fixed, thanks for pointing out!

**Pull request merged**